torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -513,6 +513,12 @@ class ShapePropagator {
         }
         return;
       }
+      case prim::unchecked_unwrap_optional: {
+	if (node->input()->type()->isSubtypeOf(TensorType::get())) {
```

Comment:
The more general case here is to set the output to be the unwrapped type of the input if it's an optional input, and the input if it's not an optional

Reply:
I think we need to be a bit more detailed than that about None. So I'd use the following:
- if it's an `IValue` and that `isNone`, leave the output alone,
- if it's an `OptionalType` return the `elementType`,
- otherwise return the Type.

Reply:
Well, for unsafe, we could always pass on the type because it should only be inserted where we know it's not None, so I'll skip the first there.